packages/core/src/context/context.ts

```diff
@@ -59,45 +62,79 @@ export const registerAll = (boundDependencies: BoundDependency<any, any>[]) => (
     context,
   );
 
+/**
+ * Resolves eager dependencies within the context
+ * @since v2.0.0
+ */
 export const resolve = async (context: Context): Promise<Context> => {
+  const resolveEagerDependency = ({ token, dependency }: BoundDependency<unknown, ContextEagerReader>): T.Task<Context> => pipe(
+    () => pipe(context, dependency.eval, d => Promise.resolve(d)),
+    T.chain(resolvedDependency => T.fromIO(() => context.set(token, resolvedDependency))));
+
   for (const [token, dependency] of context) {
-    context.set(
-      token,
-      isEagerDependency(dependency)
-        ? await Promise.resolve(dependency.eval(context))
-        : dependency,
-    );
+    if (!isEagerDependency(dependency)) continue;
```

Comment:
Can you use `isLazyDependency(dependency)`?

Reply:
What if I introduce a new type of dependency? I only want to skip dependencies that are not `eager` 🤔

Reply:
Then it should stay as it is :+1: